README.md

```diff
@@ -1,4 +1,5 @@
-# p4-constraints ![](https://github.com/p4lang/p4-constraints/workflows/build%20&%20unit%20test/badge.svg)
+![](https://github.com/p4lang/p4-constraints/workflows/build%20&%20unit%20test/badge.svg)
```

Comment:
Does this mean it will only show unit tests here, not all tests?

Reply:
Yes. I added a second badge for the gold tests

Reply:
A related point is that we could in principle do all testing in the Docker container and run only a single job (plus another one for formatting).
On the other hand, I feel like having them separate may make it easier to diagnose problems, plus we get some additional coverage since we do both a Docker build on Ubuntu 16.04 as well as a native build on "ubuntu latest" currently. Wdyt?

Reply:
Looks like the docker build is also quite a bit slower, so getting quicker feedback on build & unit tests seems useful.

Reply:
I agree, that seems good. Once we have p4c as native dependency, we could do everything without docker, right? Maybe at that point we should have 1 for native of all tests, and one for docker to make sure our docker setup works?

Reply:
Yes, agreed.
That being said, I'm starting to think the current Docker solution is actually superior to building p4c natively. The Docker image is officially provided by p4lang/p4c, so we can rely on it to always work. Having our own bazel rules (unfortunately) seems brittle as long as they are not officially supported by p4lang/p4c. That being said, they would simplify development and would also speed up builds. All I'm saying is, there is a cost to relying on bazel rules if they are not officially supported, as they are bound to break from time to time. It would be unfortunate for our tests to break regularly just because someone made a change to p4c.